src/Components/Footer.jsx

```diff
@@ -1,55 +1,151 @@
-
-        <Grid.Column
-          textAlign='center'
-          width={10}>
-
-          <Header
-            color='orange'
-            id='footer-logo'
-            as={Link}
-            to='/'>
-            Being Stockholm
-          </Header>
-
-        </Grid.Column>
-
-        <Grid.Column
-          id='profile-icon'
-          as={Link}
-          to='log-in'
-          textAlign='center'
-          width={3}>
-          <svg className='footer-icon' xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill={(this.props.currentUser.isSignedIn)? '#F2711C' : '#D3D3D3'} d="M10 20a10 10 0 1 1 0-20 10 10 0 0 1 0 20zM7 6v2a3 3 0 1 0 6 0V6a3 3 0 1 0-6 0zm-3.65 8.44a8 8 0 0 0 13.3 0 15.94 15.94 0 0 0-13.3 0z" /></svg>
-        </Grid.Column>
-
-      </Grid>
+            </Grid.Column>
+
+            <Grid.Column
+              id='profile-icon'
+              as={Link}
+              to='log-in'
+              textAlign='center'
+              width={3}>
+              <svg className='footer-icon' xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill='#F2711C' d="M10 20a10 10 0 1 1 0-20 10 10 0 0 1 0 20zM7 6v2a3 3 0 1 0 6 0V6a3 3 0 1 0-6 0zm-3.65 8.44a8 8 0 0 0 13.3 0 15.94 15.94 0 0 0-13.3 0z" /></svg>
+            </Grid.Column>
+          </Grid>
+        </>
+      )
+    } else {
+      adminIcon = (
+        <>
+          <Grid
+            id='footer'
+            verticalAlign='middle'
+            centered columns={3}>
+
+            <Grid.Column
+              textAlign='center'
+              verticalAlign='middle'
+              width={3}>
+              <svg className='footer-icon' id='footer-menu-icon' xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" onClick={() => this.props.dispatch({ type: 'CHANGE_VISIBILITY' })}><path fill='#F2711C' d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" /></svg>
+            </Grid.Column>
+
+            <Grid.Column
+              textAlign='center'
+              width={10}>
+
+              <Header
+                color='orange'
+                id='footer-logo'
+                as={Link}
+                to='/'>
+                Being Stockholm
+              </Header>
+
+            </Grid.Column>
+
+            <Grid.Column
+              id='profile-icon'
+              as={Link}
+              to='log-in'
+              textAlign='center'
+              width={3}>
+              <svg className='footer-icon' xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill='#D3D3D3' d="M10 20a10 10 0 1 1 0-20 10 10 0 0 1 0 20zM7 6v2a3 3 0 1 0 6 0V6a3 3 0 1 0-6 0zm-3.65 8.44a8 8 0 0 0 13.3 0 15.94 15.94 0 0 0-13.3 0z" /></svg>
+            </Grid.Column>
+          </Grid>
+        </>
+      )
+    }
+
+
+    return (
+      <>
+        {adminIcon}
+      </>
     )
   }
 }
 
 const mapStateToProps = (state) => {
   return {
+    admin: state.reduxTokenAuth.currentUser.attributes.admin,
     currentUser: state.reduxTokenAuth.currentUser
   }
 }
```

Comment:
id refactor this whole component that we only have a conditional on user/admin icon and leave the menu icon and being sthlm link out of it, to keep it clean and dry

Reply:
If this was another week project which we would never continue working on, I would agree with you. But others will probably work on this in the future and there might be a need to separate the footer like that. I believe we should leave it as is, even if it looks as premature at the moment.

Reply:
done

Reply:
thankkkks!!!!!!!